src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -336,8 +373,13 @@ t8_forest_adapt (t8_forest_t forest)
         elements[0] = t8_element_array_push (telements);
         /* Compute the parent of the current family.
          * This parent is now inserted in telements. */
+        T8_ASSERT (tscheme->t8_element_level (elements_from[0]) > 0);
         tscheme->t8_element_parent (elements_from[0], elements[0]);
         el_inserted++;
+        if (num_siblings > curr_size_elements) {
+          elements = T8_REALLOC (elements, t8_element_t *, num_siblings);
+          curr_size_elements = num_siblings;
```

Comment:
This is done on purpose to avoid one computation of the number of children. If you want to be super-safe, we can compute the number of children of elements[0] here but:
num_siblings is set to the size of the family considered in elements_from. Computing the parent of 'elements_from[0]' and storing it in 'elements[0]' makes 'num_siblings' equivalent to the number of children of 'elements[0]'
To make this behaviour more clear i added an update:
'num_children = 'num_siblings'
And used the varialbe 'num_children' to update the size of the array.